### PR TITLE
Implement LWG-3721: Allow an arg-id with a value of zero for width in std-format-spec

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
Towards #2872.

### CODE REVIEW

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Excellent, thank you! :heart_eyes_cat:

File: tests/std/tests/P0645R10_text_formatting_death/test.cpp [hunk 1]
```diff
@@ -17,8 +17,8 @@ void test_case_advance_no_range() {
     context.advance_to(other_format_string.begin());
 }
 
-void test_case_zero_dynamic_width() {
-    (void) format("{:{}}", 42, 0);
+void test_case_negative_dynamic_width() {
```
Comment (CaseyCarter) [inline tests/std/tests/P0645R10_text_formatting_death/test.cpp:20]: I've brought this back as a failure case for negative dynamic width. (No change requested.)

**Review by CaseyCarter — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for implementing so many LWG issue resolutions and improving conformance! :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat: :heart_eyes_cat:

**Pull request merged**